fastdds_python/test/api/test_domainparticipant.py

```diff
@@ -1130,15 +1130,33 @@ def test_create_and_delete_topic():
            factory.delete_participant(participant))
 
 
-#
-#    /**
-#     * @brief This operation enables the DomainParticipant
-#     *
-#     * @return RETCODE_OK
-#     */
-#    RTPS_DllAPI ReturnCode_t enable() override;
-#
-#    // DomainParticipant specific methods from DDS API
+def test_delete_contained_entities():
+    factory = fastdds.DomainParticipantFactory.get_instance()
+    assert(factory is not None)
+    participant = factory.create_participant(
+            0, fastdds.PARTICIPANT_QOS_DEFAULT)
+    assert(participant is not None)
+
+    publisher = participant.create_publisher(
+            fastdds.PUBLISHER_QOS_DEFAULT)
+    assert(publisher is not None)
+    subscriber = participant.create_subscriber(
+            fastdds.SUBSCRIBER_QOS_DEFAULT)
+    assert(subscriber is not None)
+    test_type = fastdds.TypeSupport(test_complete.CompleteTestTypePubSubType())
+    assert(fastdds.ReturnCode_t.RETCODE_OK ==
+           participant.register_type(test_type, test_type.get_type_name()))
+    topic = participant.create_topic(
+            "Complete", "CompleteTestType", fastdds.TOPIC_QOS_DEFAULT)
+    assert(topic is not None)
+
+    assert(fastdds.ReturnCode_t.RETCODE_OK ==
```

Comment:
I would check that `delete_participant` returns `PRECONDITION_NOT_MET` if the entities have not been deleted first